### PR TITLE
Region awareness support

### PR DESCRIPTION
**Pull request by timcharper**
We add support for region awareness in USI. USI will detect the region
of the Mesos master and default to launching pods in that region. A
generic, implementable DomainSelector is implemented with default
implementations to match the home region and to match a specific region.

JIRA Issues: DCOS_OSS-5628

### CODE REVIEW

File: core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala [hunk 1]
```diff
@@ -1,35 +1,80 @@
...
 
-/**
-  * Interface for the offer matcher implementations. Given a Mesos [[Mesos.Offer]] and a set of [[RunningPodSpec]]s
-  * return a map of all matched PodSpecs with the list of matched Mesos resources.
-  */
-trait OfferMatcher {
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+
+class OfferMatcher(masterDomainInfo: Mesos.DomainInfo) {
```
Comment (jeschkies) [inline core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala:10]: What was the motivation to move the implementation here?

File: core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/DomainSelector.scala [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.mesosphere.usi.core.models.faultdomain
+
+import org.apache.mesos.v1.Protos.DomainInfo
+
+trait DomainSelector {
```
Comment (jeschkies) [inline core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/DomainSelector.scala]: I wonder if this is rather a matcher. And if so maybe we could unify the interfaces for matching. What do you think?

File: core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala (hunk 1 above)
Comment (timcharper) [reply core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala:10]: It was premature generalization.

File: core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/DomainSelector.scala (hunk 2 above)
Comment (timcharper) [reply core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/DomainSelector.scala]: It's a special kind of matcher which can take in to consideration the Mesos domain

**Review by kaiwalyajoshi — APPROVED** [review]

**Pull request merged**